api/routes/batch/create.go

```diff
@@ -46,7 +47,7 @@ func Create(producer *kafka.Producer, admin *kafka.AdminClient, rdb *redis.Clien
 		batchID := batchUUID.String()
 
 		// List resources of current batch in Redis
-		if err := rdb.SAdd("batch:"+batchID+":resources", resourceIDs).Err(); err != nil {
+		if err := ctl.Rdb.SAdd("batch:"+batchID+":resources", resourceIDs).Err(); err != nil {
```

Comment:
you should  not be able to access the redis client of the batchController. Add a method instead